quic/s2n-quic-core/src/recovery/bbr/probe_bw.rs

```diff
@@ -0,0 +1,759 @@
+            // TODO: Update this logic based on subsequent draft updates or consider lowering
+            //       bw_hi in `on_inflight_too_high`
+            if rate_sample.bytes_in_flight as u64 > self.data_volume_model.inflight_hi() {
+                self.data_volume_model
+                    .update_upper_bound(rate_sample.bytes_in_flight as u64);
+            }
+
+            if self.probe_bw_state.cycle_phase == CyclePhase::Up
+                && self.is_congestion_limited()
+                && self.cwnd as u64 >= self.data_volume_model.inflight_hi()
+            {
+                // inflight_hi is being fully utilized, so probe if we can increase it
+                self.probe_bw_state.probe_inflight_hi_upward(
+                    bytes_acknowledged,
+                    &mut self.data_volume_model,
+                    self.cwnd,
+                    self.max_datagram_size,
+                    self.round_counter.round_start(),
+                );
+            }
+        }
+    }
+
+    /// Update AckPhase and advance the Max BW filter if necessary
+    fn update_ack_phase(&mut self, rate_sample: RateSample) {
+        // TODO: let is_probe_bw = self.state == ProbeBw
+        let is_probe_bw = true;
+
+        match self.probe_bw_state.ack_phase {
+            AckPhase::ProbeStarting => {
+                // starting to get bw probing samples
+                self.probe_bw_state.ack_phase = AckPhase::ProbeFeedback;
+            }
+            AckPhase::ProbeStopping => {
+                self.probe_bw_state.bw_probe_samples = false;
+                self.probe_bw_state.ack_phase = AckPhase::Init;
+
+                if is_probe_bw && !rate_sample.is_app_limited {
+                    self.data_rate_model.advance_max_bw_filter();
+                }
+            }
+            _ => {}
+        }
+    }
+
+    /// Called when loss indicates the current inflight amount is too high
+    pub fn on_inflight_too_high<Rnd: random::Generator>(
+        &mut self,
+        is_app_limited: bool,
+        bytes_in_flight: u32,
+        target_inflight: u32,
+        random_generator: &mut Rnd,
+        now: Timestamp,
+    ) {
+        self.probe_bw_state.bw_probe_samples = false; // only react once per bw probe
+        if !is_app_limited {
+            self.data_volume_model.update_upper_bound(
+                (bytes_in_flight as u64).max((bbr::BETA * target_inflight as u64).to_integer()),
+            )
+        }
+
+        // TODO: Check self.state == State::ProbeBw
+        if self.probe_bw_state.cycle_phase == CyclePhase::Up {
+            self.probe_bw_state.start_down(
+                &mut self.congestion_state,
+                &mut self.round_counter,
+                self.bw_estimator.delivered_bytes(),
+                random_generator,
+                now,
+            );
+        }
+    }
+
+    /// Returns true if it is time to transition from `Down` to `Cruise`
+    fn check_time_to_cruise(&self) -> bool {
```

Comment:
```suggestion
    fn is_time_to_cruise(&self) -> bool {
```

Reply:
Another name from the RFC, your name is better though. How about I comment with the RFC name?